### ISSUE TITLE
Backup to UNC or relative path

### ISSUE THREAD

**Issue by mason-ftl** [marvinweber/KPSimpleBackup#10]
Hello Marvin,

Thanks for creating this plugin!

I host my KeePass databases on a samba file server.  The databases are simultaneously accessed by multiple people.  I also host the KeePass executable, plugins and config file on a read-only share (this isn't strictly relevant to my request, but seems like potentially useful context for this request).

The documented recommendation, of saving to a mapped drive, isn't ideal as I can't guarantee that all users will have the same mapped drives.  I'd like to be able to send my backups to a different folder on that same file server.  Therefore, being able to specify a UNC path or a relative path would be ideal.

It looks like backing up to a relative path would work, as that's essentially what is done in the [SimpleDatabaseBackup](https://github.com/jnko/SimpleDatabaseBackup/blob/master/SimpleDatabaseBackup/SimpleDatabaseBackup.cs) project.  I've tried this plugin and it does save the backup files to the file server; unfortunately, they are all in the same folder as the original files.  If you decide to implement this, perhaps creating a backups folder in the same folder and then writing the backups to the new folder, would be a good way to go?

Backing up to a UNC path would probably be the best option as that would mean that I could specify a single location for all backups, regardless of whether the user opens a local database from their PC or a database on the file server.

**Comment by marvinweber** [comment]
Hey, thank you for this request :)  

Just so that I got that right: You have to use the same **backup-path** for all users, as they all use the same config file (since keepass is stored on the file share as well)?  
And if you could define a path like `\\yourserver\path\to\backupfolder` it would "point" for every user to the same folder and backups would work for everyone.

If that is the problem, I will have a look at it and try to implement it with the next feature release :)

**Comment by mason-ftl** [comment]
>
> Just so that I got that right: You have to use the same *backup-path* for
> all users, as they all use the same config file (since keepass is stored on
> the file share as well)?
>
It's not that I have to use the same backup path, it's that I really want
to use the same backup path.  I want to have one authoritative copy of each
of the databases and I want to have a record of every change that is made
to these databases.  An audit log would also be great, so that I can see
who made which change and when.  I actually managed to accomplish this,
last night, by using the triggers feature in KeePass.  It's pretty slick!
I'll post the trigger XML, so that you can see how it works.


> And if you could define a path like \\yourserver\path\to\backupfolder it
> would "point" for every user to the same folder and backups would work for
> everyone.
>
Yes


> If that is the problem, I will have a look at it and try to implement it
> with the next feature release :)
>
Awesome!

The one thing that my trigger doesn't do is clean up old backups.  I was
just going to implement that with a cronjob (the file server is Samba on
Linux).  However, if you're going to continue evolving your plugin, that
might be a better/easier approach.

>

**Comment by mason-ftl** [comment]
`<?xml version="1.0" encoding="utf-8"?>
<TriggerCollection xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<Triggers>
		<Trigger>
			<Guid>81HT2b4ZbUGziFZPXi8mEg==</Guid>
			<Name>Audit database changes</Name>
			<Events>
				<Event>
					<TypeGuid>s6j9/ngTSmqcXdW6hDqbjg==</TypeGuid>
					<Parameters>
						<Parameter>0</Parameter>
						<Parameter />
					</Parameters>
				</Event>
			</Events>
			<Conditions />
			<Actions>
				<Action>
					<TypeGuid>tkamn96US7mbrjykfswQ6g==</TypeGuid>
					<Parameters>
						<Parameter />
						<Parameter>0</Parameter>
					</Parameters>
				</Action>
				<Action>
					<TypeGuid>2uX4OwcwTBOe7y66y27kxw==</TypeGuid>
					<Parameters>
						<Parameter>%comspec%</Parameter>
						<Parameter>/c echo {DB_BASENAME} changed on %DATE% %TIME% by %USERDNSDOMAIN% %USERDOMAIN% %USERNAME% &gt;&gt; "\\fileserver\Vault\Backups\KpAuditlog.txt"</Parameter>
						<Parameter>False</Parameter>
						<Parameter>0</Parameter>
						<Parameter />
					</Parameters>
				</Action>
				<Action>
					<TypeGuid>2uX4OwcwTBOe7y66y27kxw==</TypeGuid>
					<Parameters>
						<Parameter>%comspec%</Parameter>
						<Parameter>/c copy "{DB_PATH}" "\\fileserver\Vault\Backups\{DB_BASENAME}.{DT_SIMPLE}"</Parameter>
						<Parameter>False</Parameter>
						<Parameter>0</Parameter>
						<Parameter />
					</Parameters>
				</Action>
				<Action>
					<TypeGuid>tkamn96US7mbrjykfswQ6g==</TypeGuid>
					<Parameters>
						<Parameter />
						<Parameter>1</Parameter>
					</Parameters>
				</Action>
			</Actions>
		</Trigger>
	</Triggers>
</TriggerCollection>`

**Comment by mason-ftl** [comment]
Of course, full credit goes to the KeePass website, where I found a couple trigger examples, that I was able to modify, to make this work.
